src/biokbase/narrative/tests/test_job.py

```diff
@@ -201,19 +202,23 @@ def check_job_attrs(self, job, job_id, exp_attrs=None, skip_state=False):
         if not exp_attrs and not skip_state:
             state = create_state_from_ee2(job_id)
             with mock.patch(CLIENTS, get_mock_client):
-                self.assertEqual(state, job.state())
+                self.assertEqual(state, job.refresh_state())
 
         attrs = create_attrs_from_ee2(job_id)
         attrs.update(exp_attrs)
 
-        # Mock here because job.child_jobs and job.retry_ids can
-        # cause EE2 query
-        with mock.patch(CLIENTS, get_mock_client):
-            for name, value in attrs.items():
-                self.assertEqual(value, getattr(job, name))
+        for name, value in attrs.items():
+            if name in ["child_jobs", "retry_ids"]:
+                # job.child_jobs and job.retry_ids may query EE2
+                with mock.patch(CLIENTS, get_mock_client):
+                    self.assertEqual(value, getattr(job, name))
+            else:
+                with assert_obj_method_called(
```

Comment:
make sure that only the `child_jobs` and `retry_ids` attributes trigger an ee2 query

Reply:
Magic to be axed later?